### PR TITLE
Allow /* comments between < and JSX element name

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes issue #4514

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -3183,6 +3183,7 @@ namespace ts {
                     if (sourceFile.languageVariant !== LanguageVariant.JSX) {
                         return parseTypeAssertion();
                     }
+                    debugger;
```
Comment (DanielRosenwasser) [inline src/compiler/parser.ts]: # no.

**Comment by DanielRosenwasser** [comment]
Apart from the `debugger` statement, looks good to me. :+1:

File: src/compiler/parser.ts (hunk 1 above)
Comment (RyanCavanaugh) [inline src/compiler/parser.ts]: :rage1:

Comment (weswigham) [inline src/compiler/parser.ts]: We could add a lint for this. No lies. Alongside 'no-null' we can have 'no-debugger'.

**Comment by vladima** [comment]
:+1:

File: src/compiler/parser.ts (hunk 1 above)
Comment (DanielRosenwasser) [inline src/compiler/parser.ts]: Sure, but have you ever considered how much I enjoy doing this?

File: src/compiler/scanner.ts [hunk 2]
```diff
@@ -1361,7 +1361,9 @@ namespace ts {
                         if (text.charCodeAt(pos + 1) === CharacterCodes.equals) {
                             return pos += 2, token = SyntaxKind.LessThanEqualsToken;
                         }
-                        if (text.charCodeAt(pos + 1) === CharacterCodes.slash && languageVariant === LanguageVariant.JSX) {
+                        if (text.charCodeAt(pos + 1) === CharacterCodes.slash &&
+                                text.charCodeAt(pos + 2) !== CharacterCodes.asterisk &&
+                                languageVariant === LanguageVariant.JSX) {
```
Comment (mhegazy) [inline src/compiler/scanner.ts]: i would move the `languageVariant === LanguageVariant.JSX` to the beginning of the if condition.

Comment (RyanCavanaugh) [inline src/compiler/scanner.ts]: Good point

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**